src/Main.enso

```diff
@@ -1,105 +1,193 @@
-    my_circ_2 = here.draw_circle 36 'blue' # equivalent to here.draw_circle 36 'blue' 0 0
-    my_circ_factory = here.draw_circle 36 'blue' ... # equivalent to (x=0)-> (y=0)-> here.draw_circle 36 'blue' x y
-
-my_method arg1 arg2 =
-    IO.println arg1
-    IO.println arg2
     # The last line in a block of code is the return value
-    arg1 + arg2
+    IO.println here.sum 1 99
+
+## Built-in data types
+data_types =
+    # Numbers
+    integer = 6
+    decimal = 78.23
+
+    # Text
+    text_sample = "this \n is my text"
+    text_sample_2 = 'this \n is my text'
+    text_block = '''
+        My text
+        can have multiple
+            lines
+        now
+    # `+` concatenates text
+    concat = text_sample + text_block
+
+    # Booleans
+    my_true = True
+    my_false = False
+
+    # Boolean logic operators
+
+    True && False
+    True || False
+    True.not
+
+    my_bool = 1 == 2
+    my_bool = 2 > 0.3
+    my_bool = "foo" == "bar"
+
+    # Vectors:
+    # Create with []
+    my_vec = [1, 2, 3]
+    # Vectors can contain multiple types:
+    my_vec_2 = [1, "Foo", False]
+    # `characters` is a method that creates a vector of characters in text.
+    text_sample.characters 
 
-draw_circle radius=1.0 color='red' x_position=0 y_position=0 =
-    ...
+    # Maps
+    # Key:value data structure.
 
+    empty_map = Map.empty
 
-## Type definitions
+    inserted = empty_map.insert 4 "foo" 
+    item = inserted.get 4 # returns "foo"
+    item_2 = inserted.get 10 # returns Dataflow Error 
+
+## Type definition
+# Types are defined with the `type` keyword,
+# followed by a type name and fields.
 type Geo_Point latitude longitude
 
-## Type definitions
+# Types can also be defined with default values for fields.
 type Geo_Point latitude longitude shape='star'
 
-## Defines a method on an Atom (a defined type)
+# Methods can be defined as Extension Methods, on an existing type (an Atom).
 Geo_Point.convert =
-    result = Geo_Point (this.latitude / 3600000) (this.longitude / 3600000) # `this` refers to the `Geo_Point` object
-    here.my_method result.latitude result.longitude # `here` still refers to the current module
+    # `this` refers to the `Geo_Point` object
+    result = Geo_Point (this.latitude / 3600000) (this.longitude / 3600000)
+    # `here` still refers to the current module
+    here.my_method result.latitude result.longitude
     result
 
 test_types =
-    # Create a Geo_Point
+    # Create a Geo_Point:
     my_point = Geo_Point 52.05 15.2
-    # getters!
+    # Fields can be used as getters:
     my_latitude = my_point.latitude
     my_longitude = my_point.longitude
-    # Pattern matching:
+    # Pattern matching on fields:
     my_number = case my_point of
         Geo_Point ltd lng -> ltd * lng
     # Calling a method on a value
     my_point.convert
 
-# Polyglot Java
+# Types can be nested and combined with pattern matching to provide
+# Option-like types.
```

Comment:
I know what you mean, but I don't like this phrasing – something like `union-like types`? Or just `to create sum types` – this is what they actually are.